docs/code-style.md

```diff
@@ -0,0 +1,45 @@
+# KubeAdmiral Code Style
+
+## Linting
+
+KubeAdmiral uses [golangci-lint](https://golangci-lint.run/usage/install/) for linting. All code should adhere to the rules specified in the [configuration file](../.golangci.yml).
+
+In order to check if your code adheres to our rules, you may use golangci-lint. First, ensure that you have golangci-lint installed. Then, run the following command:
+
+```console
+$ golangci-lint run --new-from-rev=origin/develop --timeout=2m --max-issues-per-linter=10 --go=1.19 --whole-files
+```
+
+## Logging Conventions
+
+Logs play an important role when troubleshooting issues. Thus, it is also important to ensure that our logs are readable, navigable and doesn't devolve into noise.
+
+In order to achieve this, KubeAdmiral follows a set of logging conventions described below.
+
+1. **ALWAYS** use [klog](https://pkg.go.dev/k8s.io/klog/v2) for logging. Klog is the de facto logging package for Kubernetes development.
+
+2. Use contextual logging whenever possible. **ALWAYS** use `klog.FromContext` to obtain a logger whenever there is a `Context` available.
+
+3. Consequently, top-level controllers or reconciliation workers **MUST** inject a logger into its `Context` using `klog.NewContext` before passing it to any nested calls. The injected logger **MUST** have the following values:
+
+    1. `controller`: indicating the name of the controller
```

Comment:
There's an alternative style present in the code: using `klog.LoggerWithName(klog.Background, controllerName)`. Do we prefer injecting `controller` as a key or setting the logger name?

Reply:
Good catch, I personally prefer using `klog.LoggerWithName(klog.Background, controllerName)`. @SOF3  what do you think?

Reply:
How would we handle FTC subcontrollers? With normal keys, it could be just `controller=scheduler` `controllerFtc=deployments.apps`, but with logger names, that might mean we have multiple dimensions in the same key.

Reply:
I prefer separating the subcontroller name and FTC name into separate keys like @SOF3 suggested, the prefixed controller names had always been a headache in the past.

Reply:
Let's go with original convention (having the controller name as a field) specified by this rule then. I added an extra `ftc` value to record the name of the corresponding FTC.